clang/lib/Sema/SemaDecl.cpp

```diff
@@ -14690,6 +14690,14 @@ void Sema::FinalizeDeclaration(Decl *ThisDecl) {
     }
   }
 
+  // The result of __builtin_counted_by_ref cannot be assigned to a variable.
+  // It allows leaking and modification of bounds safety information.
+  if (const auto *CE = dyn_cast_if_present<CallExpr>(VD->getInit());
+      CE && CE->getBuiltinCallee() == Builtin::BI__builtin_counted_by_ref)
```

Comment:
Can you not use `IsBuiltinCountedByRef()` here?

Reply:
Done.